object-storage-core/src/main/java/io/micronaut/objectstorage/request/FileUploadRequest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.objectstorage.request;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.objectstorage.ObjectStorageException;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URLConnection;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+/**
+ * Upload request implementation using {@link java.io.File}.
+ */
+public class FileUploadRequest implements UploadRequest {
+    private final String keyName;
```

Comment:
I don't think we can have immutable `keyName` and `contentType.` 

If you are uploading a file to an S3 bucket with:

```java
 public HttpResponse<String> upload(CompletedFileUpload fileUpload) {
         UploadRequest objectStorageUpload = UploadRequest.fromCompletedFileUpload(fileUpload);
```

Suppose two users upload a PDF named `A.pdf` to an S3 bucket. The second user upload will override the first user's file. You probably want to generate a unique UUID for the key and maybe use the original filename in the metadata of the upload. 

Moreover, my experience with the content type from complete file upload is that sometimes you want to define it manually. 

I think `keyName` and `contentType` should have setters.

Reply:
I think it maybe even better to force users to supply to keyName: 

instead of 

```java
static UploadRequest fromCompletedFileUpload(CompletedFileUpload completedFileUpload) {
```

Maybe


```java
`static UploadRequest fromCompletedFileUpload(String keyName, CompletedFileUpload completedFileUpload) {
```